src/libraries/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -426,16 +453,32 @@ public void PathWithReservedDeviceNameAsExtendedPath(string path)
         [Theory,
             MemberData(nameof(UncPathsWithoutShareName))]
         [PlatformSpecific(TestPlatforms.Windows)]
-        public void UncPathWithoutShareNameAsPath_ThrowsIOException_Core(string path)
+        public void UncPathWithoutShareNameAsPath_ThrowsIOException(string path)
         {
-            Assert.ThrowsAny<IOException>(() => Create(path));
+            // Same exception message, different exception type
+            if (PlatformDetection.IsNetCore)
+            {
+                Assert.ThrowsAny<IOException>(() => Create(path));
+            }
+            else
+            {
+                Assert.Throws<ArgumentException>(() => Create(path));
+            }
         }
 
         [Fact]
         [PlatformSpecific(TestPlatforms.Windows)]
-        public void UNCPathWithOnlySlashes_Core()
+        public void UNCPathWithOnlySlashes()
         {
-            Assert.ThrowsAny<IOException>(() => Create("//"));
+            // Same exception message, different exception type
+            if (PlatformDetection.IsNetCore)
+            {
+                Assert.ThrowsAny<IOException>(() => Create("//"));
+            }
+            else
+            {
+                Assert.Throws<ArgumentException>(() => Create("//"));
+            }
```

Comment:
Same question.  Is there a reason they need to be different?  If not, let's try to be consistent.  Thanks.